nixos/modules/config/users-groups.nix

```diff
@@ -119,7 +119,7 @@ let
       };
 
       shell = mkOption {
-        type = types.either types.shellPackage types.path;
+        type = types.nullOr (types.either types.shellPackage types.path);
```

Comment:
Question: If I set a shell in nixos, does this override the user shell on every update?

Reply:
My understanding of nixos/modules/config/update-users-groups.pl is that:
- it always sets the `shell`
- if a given user has a shell set in the NixOS config, that will always be used, overwriting any existing shell

Reply:
For what it's worth, this isn't changing the existing behaviour at all, right? It just allows you to additionally explicitly set a user's shell to null which then means NixOS won't touch it.

Reply:
Yep, that's correct!